### PR TITLE
fix: missing go back button on squads page

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Refactor the component to navigate to be used by other components.
- Add the missing back button on Squads page.

Screenshot 2024-04-02 at 10 13 25 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-262 #done

### Preview domain
https://mi-262.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/post/GoBackHeaderMobile.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React, { PropsWithChildren, ReactElement } from 'react';
```
Comment (sshanzel) [inline packages/shared/src/components/post/GoBackHeaderMobile.tsx:1]: This was just a renamed file.

**Comment by sshanzel** [comment]
Devcard was also missing it, so I've included the fix here along the missing footer.

File: packages/webapp/pages/squads/[handle]/index.tsx [hunk 2]
```diff
@@ -211,7 +211,7 @@ const SquadPage = ({
 };
 
 SquadPage.getLayout = getLayout;
-SquadPage.layoutProps = mainFeedLayoutProps;
+SquadPage.layoutProps = { ...mainFeedLayoutProps, canGoBack: true };
```
Comment (nensidosari) [inline packages/webapp/pages/squads/[handle]/index.tsx:214]: Did only the squad page and dev card need this addition?

**Review by nensidosari — APPROVED** [review]
Looks all good, just one check

File: packages/webapp/pages/squads/[handle]/index.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/pages/squads/[handle]/index.tsx:214]: Yep. Based from their feedback.

**Pull request merged**